LX/src/main/java/heronarts/lx/LXSnapshot.java

```diff
@@ -0,0 +1,243 @@
+package heronarts.lx;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import heronarts.lx.modulator.LXModulator;
+import heronarts.lx.parameter.CompoundParameter;
+import heronarts.lx.parameter.LXParameter;
+
+public class LXSnapshot extends LXComponent {
```

Comment:
I think the user should be able to rename these like patterns, etc. Get this with:

implements LXComponent.Renamable

Reply:
Renaming actually works from the ui, not sure why if it should require this interface.

Reply:
https://github.com/heronarts/LX/blob/master/LX/src/main/java/heronarts/lx/LXComponent.java#L660

It won't be properly restored on the next run. This is to prevent people from going rogue and renaming things like the core Engine, Modulation Engines, Audio Engine, etc.